examples/order_posting_e2e.py

```diff
@@ -0,0 +1,120 @@
+ORDER_KIND = "sell"
+CHAIN = Chain.SEPOLIA
+CHAIN_ID = SupportedChainId.SEPOLIA
+
+config = OrderBookAPIConfigFactory.get_config("prod", CHAIN_ID)
+ORDER_BOOK_API = OrderBookApi(config)
+
+ADDRESS = os.getenv("USER_ADDRESS")
+ACCOUNT = Account.from_key(os.getenv("PRIVATE_KEY"))
+
+
+async def get_order_quote(
+    order_quote_request: OrderQuoteRequest, order_side: OrderQuoteSide1
+) -> OrderQuoteResponse:
+    return await ORDER_BOOK_API.post_quote(order_quote_request, order_side)
+
+
+def sign_order(order: Order) -> EcdsaSignature:
+    order_domain = asdict(
+        domain(
+            chain=CHAIN, verifying_contract=CowContractAddress.SETTLEMENT_CONTRACT.value
+        )
+    )
+    del order_domain["salt"]  # TODO: improve interfaces
+
+    return _sign_order(order_domain, order, ACCOUNT, SigningScheme.EIP712)
+
+
+async def post_order(order: Order, signature: EcdsaSignature) -> UID:
+    order_creation = OrderCreation(
+        **{
+            "from": ADDRESS,
+            "sellToken": order.sellToken,
+            "buyToken": order.buyToken,
+            "sellAmount": order.sellAmount,
+            "feeAmount": order.feeAmount,
+            "buyAmount": order.buyAmount,
+            "validTo": order.validTo,
+            "kind": order.kind,
+            "partiallyFillable": order.partiallyFillable,
+            "appData": order.appData,
+            "signature": signature.data,
+            "signingScheme": "eip712",
+            "receiver": order.receiver,
+        },
+    )
+    return await ORDER_BOOK_API.post_order(order_creation)
+
+
+async def main():
+    order_quote_request = OrderQuoteRequest(
+        **{
+            "sellToken": SELL_TOKEN,
+            "buyToken": BUY_TOKEN,
+            "from": ADDRESS,
+        }
+    )
+    order_side = OrderQuoteSide1(
+        kind=OrderQuoteSideKindSell.sell,
+        sellAmountBeforeFee=TokenAmount(SELL_AMOUNT_BEFORE_FEE),
+    )
+
+    order_quote = await get_order_quote(order_quote_request, order_side)
+
+    order_quote_dict = json.loads(order_quote.quote.model_dump_json(by_alias=True))
+    order = Order(
+        **{
+            "sellToken": SELL_TOKEN,
+            "buyToken": BUY_TOKEN,
+            "receiver": ADDRESS,
+            "validTo": order_quote_dict["validTo"],
+            "appData": "0x0000000000000000000000000000000000000000000000000000000000000000",
+            "sellAmount": SELL_AMOUNT_BEFORE_FEE,  # Since it is a sell order, the sellAmountBeforeFee is the same as the sellAmount
+            "buyAmount": order_quote_dict["buyAmount"],
+            "feeAmount": "0",  # CoW Swap does not charge fees
+            "kind": ORDER_KIND,
+            "sellTokenBalance": "erc20",
+            "buyTokenBalance": "erc20",
+        }
+    )
+
+    signature = sign_order(order)
+    order_uid = await post_order(order, signature)
+    print(f"order posted on link: {ORDER_BOOK_API.get_order_link(order_uid)}")
+
+
+if __name__ == "__main__":
+    asyncio.run(main())
```

Comment:
this still functional after all those refactors @yvesfracari ?

Reply:
It should be, I tested yesterday. Did you have any problem running it?